### PR TITLE
fix: Use Containerfile instead s2i

### PR DESCRIPTION
**Pull request by mateusoliveira43**
- [x] I have read [Pelorus Contributing guidelines](https://github.com/dora-metrics/pelorus/blob/master/CONTRIBUTING.md)

## Linked Issues

related to #862

## Description

Use Containerfile instead of s2i to create exporters images.

### Test in OpenShift

Create Pelorus instance with
```yaml
kind: Pelorus
apiVersion: charts.pelorus.dora-metrics.io/v1alpha1
metadata:
  name: test-containerfile
  namespace: test-pelorus-operator
spec:
  exporters:
    global: {}
    instances:
      - app_name: committime-exporter
        exporter_type: committime
        source_url: https://github.com/mateusoliveira43/pelorus
        source_ref: fix/use-containerfile-instead-s2i
```
using operator created by CI commented in this PR.

Check the logs from exporter build Pod and run `skopeo --help` in the terminal of running exporter Pod.

### Tests with E2E 

Check if e2e tests used Containerfile to create images.

For example, run `./scripts/run-pelorus-e2e-tests.sh -o konveyor` and check the logs from exporters build Pod and run `skopeo --help` in the terminal of running exporters Pod.

### After PR closes

Check if CI of following PRs work as expected:
- test images are created successfully
- exporters images are created successfully

### CODE REVIEW

File: container/Containerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# File generated by source to image (s2i v1.3.6) executable, by running
```
Comment (mateusoliveira43) [inline container/Containerfile:1]: add this to docs

File: docs/Development.md [hunk 2]
```diff
@@ -253,11 +253,30 @@ If not defined specifically, exporters are using pre-built container images with
 
 #### Pre-built Quay images
 
-Each Pelorus GitHub pull request that is [merged](https://github.com/dora-metrics/pelorus/pulls?q=is%3Apr+is%3Amerged) results in a new set of images that are tagged with the GitHub commit hash, for example `d6f6e6fa1c9d48ca1deeaf1c72585b94964cbf31` for the following [Pull Request](https://github.com/dora-metrics/pelorus/commit/d6f6e6fa1c9d48ca1deeaf1c72585b94964cbf31). The newest merged commit results in additional image tag `latest`.
+Each pull request involving `exporters` folder that is merged results in a new set of images that are tagged with the commit hash (for example, `d6f6e6fa1c9d48ca1deeaf1c72585b94964cbf31` for the following [Pull Request](https://github.com/dora-metrics/pelorus/commit/d6f6e6fa1c9d48ca1deeaf1c72585b94964cbf31)) and with the exporters version (for example, `v2.0.10-rc.5`). The newest merged commit results in additional image tag `latest`. TODO test if this does not break new version script
```
Comment (mateusoliveira43) [inline docs/Development.md:256]: TODO

File: docs/Development.md [hunk 3]
```diff
@@ -339,7 +358,7 @@ exporters:
...
+By specifying `source_url` and optionally `source_ref` Pelorus exporters will use installation method that performs incremental builds of the exporter images using source from the GIT repository. Images are being stored in an OpenShift Container Platform registry and used during Pelorus Helm deployment or update. Each instance that uses this method results in a new build. This method is recommended for development or unmerged bug-fixes as it may point to any GIT and any branch or GIT reference. By default `source_ref` points to the latest [released](https://github.com/dora-metrics/pelorus/releases) Pelorus. TODO is not master?
```
Comment (mateusoliveira43) [inline docs/Development.md:361]: TODO

**Comment by mateusoliveira43** [comment]
/hold 
discuss before merging

File: .github/workflows/build-pelorus.yml [hunk 4]
```diff
@@ -1,23 +1,39 @@
 name: Pelorus Exporter Build
 
 on:
-  push:
+  # push:
```
Comment (mateusoliveira43) [inline .github/workflows/build-pelorus.yml:4]: TODO

File: container/Containerfile [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    chown -R 1001:0 $WORK_DIR
+
+USER 1001
+
+WORKDIR $WORK_DIR
+
+RUN pip install -r requirements.txt && \
+    pip install -e .
+
+# TODO use array syntax
```
Comment (mateusoliveira43) [inline container/Containerfile:27]: TODO

File: exporters/Containerfile [hunk 6]
```diff
@@ -0,0 +1,27 @@
+FROM registry.access.redhat.com/ubi8/python-39
```
Comment (mateusoliveira43) [inline exporters/Containerfile:1]: use fixed TAG?

File: exporters/Containerfile [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+ENV APP_FILE=deploytime/app.py \
+    PELORUS_IMAGE_TAG=$PELORUS_IMAGE_TAG \
+    OPENSHIFT_BUILD_SOURCE=$OPENSHIFT_BUILD_SOURCE \
+    OPENSHIFT_BUILD_REFERENCE=$OPENSHIFT_BUILD_REFERENCE
+
+USER root
+
+COPY ./ $WORK_DIR
+
+RUN dnf -y install skopeo && \
```
Comment (mateusoliveira43) [inline exporters/Containerfile:17]: Do we need to clean cache in this layer? In debian images I usually run `rm -rf /var/lib/apt/lists/*` after `apt-get install`

Comment (mpryc) [reply exporters/Containerfile:17]: yes, any cleanup done in this step will reduce it's size. But it needs to be within line 17.

File: exporters/Containerfile (hunk 6 above)
Comment (mateusoliveira43) [reply exporters/Containerfile:1]: did not like the available options, going to stick with latest
Screenshot from 2023-05-31 16-58-47

File: .github/workflows/build-pelorus.yml [hunk 8]
```diff
@@ -26,81 +42,44 @@ jobs:
         uses: "./.github/workflow_templates/build_push_exporters"
         with:
           image: 'pelorus-committime-exporter'
-          builder_image: ${{ env.builder_image }}
-          s2i_log_level: ${{ env.s2i_log_level }}
-          quay_imageregistry: ${{ env.imageregistry }}
-          quay_imagenamespace: ${{ secrets.QUAY_IMAGE_NAMESPACE }}
+          tags: ${{ needs.get-tags.outputs.tags }}
+          # quay_imagenamespace: ${{ secrets.QUAY_IMAGE_NAMESPACE }}
```
Comment (mateusoliveira43) [inline .github/workflows/build-pelorus.yml:46]: TODO

File: .github/workflow_templates/build_push_exporters/action.yml [hunk 9]
```diff
@@ -19,35 +16,34 @@ inputs:
...
+    #     registry: quay.io
 
-    - name: Tag exporter image with additional version tag
+    - name: Build exporter image
       shell: bash
+      working-directory: ./exporters
       run: |
-        docker tag ${{ steps.build-exporter.outputs.image }}:${{ github.sha }} ${{ steps.build-exporter.outputs.image }}:${{ env.VERSION }}
+        docker image build ${{ env.PARSED_TAGS }} \
+        --build-arg PELORUS_IMAGE_TAG="${{ env.FULL_IMAGE_NAME }}:$(echo ${{ inputs.tags }} | grep -oP 'v\d+\.\d+.\d+(-rc.\d+)?')" \
```
Comment (mateusoliveira43) [inline .github/workflow_templates/build_push_exporters/action.yml:40]: this will break in test images job :(

**Review by mpryc — COMMENTED** [review]
/lgtm

**Review by mpryc — APPROVED** [review]

**Pull request merged**